ubireader/ubifs/nodes.py

```diff
@@ -160,7 +160,9 @@ def __init__(self, buf):
             setattr(self, key, fields[key])
 
         idxs = UBIFS_IDX_NODE_SZ
-        brs = UBIFS_BRANCH_SZ
+        # Dynamic detection of `brs` (branch size), whether a hash is present after
+        # the key in authenticated UBIFS. Not checking the len of the hash.
+        brs = int((len(buf) - UBIFS_IDX_NODE_SZ) / self.child_cnt)
```

Comment:
The `int` casting is hiding low-level details. Do you want to round up or round down ? It would be cleaner with `math.floor` or `math.ceil`, depending on implementation details. You can also use `//` instead of `/` if you want to round down.

Reply:
I think `//` is what I was trying to achieve. In theory, if the UBIFS image is correctly formatted, the division by `self.child_cnt` has no remainder. My `int( ... )` was a clumsy attempt to get rid of the `.0` in the result of the division. (Because `brs` needs to be an int, not a float.)

I'll use `//` instead. I'll also rewrite the comment, because I'm having a hard time to understand myself after a few months.